### PR TITLE
test - fixes createSales ans updateSales

### PR DESCRIPTION
**Pull request by wojteknowacki**
Fix for failing sales test - new helper added to ensure variant is on sale or not - depending on the scenario. 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [x] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=3

### CODE REVIEW

**Review by szczecha — APPROVED** [review]

File: cypress/support/api/utils/discounts/salesUtils.js [hunk 1]
```diff
@@ -15,42 +16,61 @@ export function createSaleInChannel({
...
-    value
+    value,
   })
     .then(saleResp => {
       sale = saleResp;
       addChannelToSale(sale.id, channelId, discountValue);
     })
     .then(() => sale);
 }
+export function shouldVariantBeOnSale(
```
Comment (krzysztofzuraw) [inline cypress/support/api/utils/discounts/salesUtils.js:33]:
```suggestion
export function getVariantWithSale({
	variantId,
	channelSlug,
	shouldBeSales,
	retries
})
```

File: cypress/support/api/utils/discounts/salesUtils.js [hunk 2]
```diff
@@ -15,42 +16,61 @@ export function createSaleInChannel({
...
+  variantId,
+  channelSlug,
+  shouldBeSales,
+  retries = 0,
+) {
+  return getVariant(variantId, channelSlug).then(salesResponse => {
+    if (salesResponse.pricing.onSale === shouldBeSales) {
+      return;
```
Comment (krzysztofzuraw) [inline cypress/support/api/utils/discounts/salesUtils.js:41]: **question**: you don't return anything here but you chain cypress `.its` to response?

File: cypress/support/api/utils/discounts/salesUtils.js (hunk 1 above)
Comment (wojteknowacki) [reply cypress/support/api/utils/discounts/salesUtils.js:33]: I thought about that - but I use it also to confirm that variant is no longer on Sale.

File: cypress/support/api/utils/discounts/salesUtils.js (hunk 2 above)
Comment (wojteknowacki) [reply cypress/support/api/utils/discounts/salesUtils.js:41]: This retrun is to break "loop" and I retrun response from 39 line

Comment (wojteknowacki) [reply cypress/support/api/utils/discounts/salesUtils.js:41]: This retrun is to break "loop" and I retrun response from 39 line

File: cypress/support/api/utils/discounts/salesUtils.js (hunk 1 above)
Comment (wojteknowacki) [reply cypress/support/api/utils/discounts/salesUtils.js:33]: confirmVariantSaleStatus - maybe this is more descriptive?

Comment (krzysztofzuraw) [reply cypress/support/api/utils/discounts/salesUtils.js:33]: what about `getVariantWithStatus({status: 'onSale'})`?

**Review by karola312 — DISMISSED** [review]

**Review by szczecha — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**